File: 1513NumberofSubstringsWithOnly1s/NumberofSubstringsWithOnly1s.js
/**
 * @param {string} s
 * @return {number}
 */
var numSub = function(s) {
    let total=0,ans=0,mod=Math.pow(10,9)+7;
    for(let i=0;i<s.length;i++){
        if(s.charAt(i)==='0'){
            ans+=(total*(total+1)/2)%mod;
            total=0
        }else{
            total++;
        }
    }
    ans+=(total*(total+1)/2)%mod;
    return ans

};

File: 1513NumberofSubstringsWithOnly1s/NumberofSubstringsWithOnly1s.java

public class NumberofSubstringsWithOnly1s {
    private class Solution{
     public int numSub(String s) {
          long ans=0,total=0,mod=1000000007L;
          for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='1'){
                total++;
            }else{
                ans = (ans + ((total * (total + 1)) / 2) % mod) % mod;
                total=0;
            }
          }

         ans = (ans + ((total * (total + 1)) / 2) % mod) % mod; 
          return (int)ans;
      }   
    }
}
